include/kernel.h

```diff
@@ -5217,6 +5217,13 @@ static inline void k_cpu_idle(void)
  * Similar to k_cpu_idle(), but called with interrupts locked if operations
  * must be done atomically before making the CPU idle.
  *
+ * Enabling interrupts and entering a low-power mode will be atomic,
```

Comment:
Comments:
1. This sentence is true for ARM. Is it true for other architectures?
2. This sentence is also true for ARM in k_cpu_idle(), as well, so this is no different from k_cpu_idle.

AFAIU, the only differences between the two functions are
a) k_cpu_atomic_idle(key) is expected to restore interrupt state to "key"
b) k_cpu_atomic_idle() is expected to be called with interrupts locked.

Reply:
The behavior requirements that I mutated from [arch_cpu_atomic_idle]( https://github.com/zephyrproject-rtos/zephyr/blob/fd6596d53bcdc4ad34edcc4a2c80a32f7e68b689/include/sys/arch_interface.h#L163) are:

> The requirements for arch_cpu_atomic_idle() are as follows:
> 1. Enabling interrupts and entering a low-power mode needs to be atomic, i.e. there should be no period of time where interrupts are enabled before the processor enters a low-power mode.  See the comments in k_lifo_get(), for example, of the race condition that occurs if this requirement is not met.
> 2. After waking up from the low-power mode, the interrupt lockout state must be restored as indicated in the 'key' input parameter.

If my translation doesn't accurately reflect the required behavior, please let me know how it should change.